impress-org/givewp
# Refactor: split gateways settings in tabs

## Description

This pull request adds tabs to the Gateways Settings page for configuring v2 and v3 payment gateways separately. Some of the legacy gateway functions have been modified to include a `$version` parameter. Going forward, v2 settings will continue to be saved in their original meta_keys, while v3 settings will be saved in meta_keys with the suffix `_v3`. These functions will now look for the correct meta_key accordingly.

## Affects

- Gateways settings page
- v2 and v3 payment gateways (enabled, label, sort, default)

## Visuals

![CleanShot 2023-09-07 at 21 26 39](https://github.com/impress-org/givewp/assets/3921017/34b0b875-79c1-467e-b56d-bd86c4d66f6d)

![CleanShot 2023-09-07 at 21 28 07](https://github.com/impress-org/givewp/assets/3921017/1c9363ed-b1f4-4297-9cd1-7e083caebe94)
_v2 form_

![CleanShot 2023-09-07 at 21 29 11](https://github.com/impress-org/givewp/assets/3921017/407ee8d4-fb57-4538-a113-f89f94f4f312)
_v3 form_

## Testing Instructions

- Open the Payment Gateways settings
- Edit v2 and v3 differently
- Open v2 and 3 forms and confirm the selected options are being reflected

## Pre-review Checklist






---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205452836799661
